# Data Exploration
Lets try read the data in and see whats in it

In [3]:
# Lets try load in the node data
with open(f'./data/cora/node', mode='r') as file:
    for i in range(10):
        lines = file.readline()
        print(lines)

31336	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	1	0	0	0	0	0	0	1	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	1	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	1	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	1	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	1	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	

In [18]:
# It seems like we can pandas read this with space seperator
import pandas as pd

nodePd = pd.read_csv('./data/cora/node', delim_whitespace=True, header=None)
# nodePd.head(10)
nodePd.rename(columns={0:'node_id', 1434:'node_class_label'}, inplace=True)
nodePd

node_id  1  2  3  4  5  6  7  8  9  ...  1425  1426  1427  1428  1429  \
0       31336  0  0  0  0  0  0  0  0  0  ...     0     0     1     0     0   
1     1061127  0  0  0  0  0  0  0  0  0  ...     0     1     0     0     0   
2     1106406  0  0  0  0  0  0  0  0  0  ...     0     0     0     0     0   
3       13195  0  0  0  0  0  0  0  0  0  ...     0     0     0     0     0   
4       37879  0  0  0  0  0  0  0  0  0  ...     0     0     0     0     0   
...       ... .. .. .. .. .. .. .. .. ..  ...   ...   ...   ...   ...   ...   
2703  1128975  0  0  0  0  0  0  0  0  0  ...     0     0     0     0     0   
2704  1128977  0  0  0  0  0  0  0  0  0  ...     0     0     0     0     0   
2705  1128978  0  0  0  0  0  0  0  0  0  ...     0     0     0     0     0   
2706   117328  0  0  0  0  1  0  0  0  0  ...     0     0     0     0     0   
2707    24043  0  0  0  0  0  0  0  0  0  ...     0     0     0     0     0   

      1430  1431  1432  1433        node_class_label  
0        0     0     0     0         Neural_Networks  
1        0     0     0     0           Rule_Learning  
2        0     0     0     0  Reinforcement_Learning  
3        0     0     0     0  Reinforcement_Learning  
4        0     0     0     0   Probabilistic_Methods  
...    ...   ...   ...   ...                     ...  
2703     0     0     0     0      Genetic_Algorithms  
2704     0     0     0     0      Genetic_Algorithms  
2705     0     0     0     0      Genetic_Algorithms  
2706     0     0     0     0              Case_Based  
2707     0     0     0     0         Neural_Networks  

[2708 rows x 1435 columns]

In [21]:
# How many labels we have
print(nodePd.loc[:,'node_class_label'].unique())
print(f"Total of number of labels: {len(nodePd.loc[:,'node_class_label'].unique())}")

['Neural_Networks' 'Rule_Learning' 'Reinforcement_Learning'
 'Probabilistic_Methods' 'Theory' 'Genetic_Algorithms' 'Case_Based']
Total of number of labels: 7


In [4]:
# Load in the link
# Lets try load in the node data
with open(f'./data/cora/link', mode='r') as file2:
    for i in range(10):
        lines = file2.readline()
        print(lines)

35	1033

35	103482

35	103515

35	1050679

35	1103960

35	1103985

35	1109199

35	1112911

35	1113438

35	1113831



In [29]:
linkPd = pd.read_csv('./data/cora/link', delim_whitespace=True, header=None)
linkPd.rename(columns={0:'citing_id', 1:'cite_from_id'}, inplace=True)
linkPd

citing_id  cite_from_id
0            35          1033
1            35        103482
2            35        103515
3            35       1050679
4            35       1103960
...         ...           ...
5424     853116         19621
5425     853116        853155
5426     853118       1140289
5427     853155        853118
5428     954315       1155073

[5429 rows x 2 columns]

The receive and give are the node ID (not index)

If there's a publication A cites publication B. Then there is a link A to B.
7 different publication Labels
2708 publication 
5429 citations between the publication

Loading None dataset...


FileNotFoundError: P5/data/cora/node not found.